src/containers/Application/ApplicationHome.tsx

```diff
@@ -1,21 +1,87 @@
-import React from 'react'
-import { FullStructure } from '../../utils/types'
+import React, { useEffect } from 'react'
+import { Button, Divider, Header, Message, Segment, Sticky } from 'semantic-ui-react'
+import { FullStructure, SectionAndPage, StageAndStatus, TemplateDetails } from '../../utils/types'
 import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
+import { ApplicationHeader, Loading } from '../../components'
+import strings from '../../utils/constants'
+import { useUserState } from '../../contexts/UserState'
+import SectionsProgress from '../../components/Sections/SectionsProgress'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { ApplicationStatus } from '../../utils/generated/graphql'
 
 interface ApplicationProps {
   structure: FullStructure
+  template: TemplateDetails
 }
 
-const ApplicationHome: React.FC<ApplicationProps> = ({ structure }) => {
+const ApplicationHome: React.FC<ApplicationProps> = ({ structure, template }) => {
+  const {
+    query: { serialNumber },
+    push,
+    replace,
+  } = useRouter()
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
   const { error, fullStructure } = useGetFullApplicationStructure({
     structure,
   })
 
-  console.log('FULL STRUCTURE', fullStructure)
-  console.log('responsesByCode', fullStructure?.responsesByCode)
-  console.log('isLoading', !fullStructure)
+  useEffect(() => {
+    if (!fullStructure) return
+    const { status } = fullStructure.info.current as StageAndStatus
+    if (status !== ApplicationStatus.Draft && status !== ApplicationStatus.ChangesRequired)
+      replace(`/application/${serialNumber}/summary`)
+  }, [fullStructure])
+
+  const handleResumeClick = ({ sectionCode, pageName }: SectionAndPage) => {
+    push(`/applicationNEW/${serialNumber}/${sectionCode}/${pageName}`)
+  }
+
+  const handleSummaryClicked = () => {
+    push(`/applicationNEW/${serialNumber}/summary`)
+  }
+
+  if (!fullStructure || !fullStructure.responsesByCode) return <Loading />
+
+  const isCompleted = Object.values(fullStructure.sections).every(
```

Comment:
Can't help thinking that this might be better calculated and stored during `generateProgress`. Do you know if this is required elsewhere? If so, probably better to do it during the hook. If not, doesn't really matter.

Reply:
Hmmm it would be required in the ApplicationPage as well I think - if we want to display the **Review & Submit** button with different appearance when the application has all required elements completed.

Reply:
Actually, we don't need to re-calculate this at all. We just need to check `fullStructure.info.firstIncompletePage` -- if it's `null`, then the application is "completed" :)

Reply:
I agree

Reply:
Yep

Reply:
Done: 7068dd3